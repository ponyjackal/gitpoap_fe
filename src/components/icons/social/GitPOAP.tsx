import { forwardRef, Ref, SVGAttributes } from 'react';
import { Icon } from './BaseIcon';

export const GitPOAP = forwardRef(
  (props: SVGAttributes<SVGSVGElement>, ref: Ref<SVGSVGElement>) => (
    <Icon
      xmlns="http://www.w3.org/2000/svg"
      width="15"
      height="17"
      fill="none"
      viewBox="0 0 15 17"
      ref={ref}
      {...props}
    >
      <path
        fill="#838495"
        fillRule="evenodd"
        d="M12.383 11.144l1.691 3.092c.176.291 0 .496-.408.496l-1.488.058c-.32 0-.73.292-.846.554l-.788 1.46a.314.314 0 01-.583 0s-1.05-1.926-1.692-3.18c.166-.048.293-.115.397-.17l.07-.035 3.646-2.275zM.539 14.236l1.692-3.092 3.617 2.275c.054.022.104.047.156.074.089.046.182.094.31.13a173.077 173.077 0 01-1.691 3.18.315.315 0 01-.584 0l-.787-1.459c-.117-.291-.525-.554-.847-.554l-1.487-.058c-.438 0-.496-.234-.38-.496zM13.345 8.43c0 .817-.496 1.663-1.225 2.072l-3.588 1.984a2.81 2.81 0 01-1.226.292c-.408 0-.846-.059-1.225-.292l-3.588-2.013c-.73-.408-1.225-1.254-1.225-2.071V4.376c0-.817.496-1.663 1.225-2.071L6.081.29C6.43.116 6.869 0 7.306 0c.409 0 .788.058 1.226.291l3.588 2.013c.787.41 1.225 1.196 1.225 2.071V8.43zm-6.943 2.393c.291.175.583.204.904.204.32 0 .641-.059.904-.204l2.597-1.459c.554-.32.904-.904.904-1.517V4.93c0-.671-.321-1.225-.904-1.517L8.21 1.954c-.32-.175-.583-.204-.904-.204-.321 0-.642.088-.904.204L3.805 3.413c-.554.32-.904.904-.904 1.517v2.917c0 .613.379 1.226.904 1.517l2.597 1.459zm-.513-5.694a.365.365 0 00-.626-.249L4.214 5.96a.365.365 0 00-.003.504l1.05 1.109a.365.365 0 00.528-.502l-.808-.854.805-.829a.365.365 0 00.103-.259zm3.052-.331a.364.364 0 00-.118.594l.825.825-.828.85a.365.365 0 10.522.51l1.079-1.109a.365.365 0 00-.003-.512l-1.08-1.08a.364.364 0 00-.397-.078zm-.77.047a.365.365 0 00-.12-.078h.001a.364.364 0 00-.475.201l-.933 2.305a.364.364 0 10.676.274l.933-2.305a.365.365 0 00-.082-.397z"
        clipRule="evenodd"
      ></path>
    </Icon>
  ),
);
