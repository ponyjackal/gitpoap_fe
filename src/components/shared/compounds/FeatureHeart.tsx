import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { Loader } from '@mantine/core';
import { FaHeart } from 'react-icons/fa';
import { ExtraHover, ExtraPressed, TextAccent, TextLight } from '../../../colors';
import { useFeaturedPOAPs, useFeaturedPOAPsDispatch } from '../../profile/FeaturedPOAPsContext';

type Props = {
  className?: string;
  poapTokenId: string;
};

const HeartWrapper = styled.div<{ isFeatured: boolean }>`
  > svg {
    fill: ${({ isFeatured }) => (isFeatured ? TextAccent : TextLight)};
  }
`;

const FeatureHeartStyled = styled(FaHeart)`
  color: ${TextLight};
  height: ${rem(18)};
  width: ${rem(18)};
  transition: 150ms fill ease;
  fill: ${TextLight};

  &:hover {
    fill: ${ExtraHover};
    cursor: pointer;
  }
  &:active {
    fill: ${ExtraPressed};
  }
`;

export const FeatureHeart = ({ className, poapTokenId }: Props) => {
  const {
    featuredPOAPsState: { featuredPOAPTokenIDs },
    showHearts,
    loadingIds,
  } = useFeaturedPOAPs();
  const { addFeaturedPOAP, removeFeaturedPOAP } = useFeaturedPOAPsDispatch();
  const isFeatured: boolean = !!featuredPOAPTokenIDs && !!featuredPOAPTokenIDs[poapTokenId];
  const isFeaturedLoading = !!loadingIds && !!loadingIds[poapTokenId];

  if (!showHearts) {
    return null;
  }

  return (
    <HeartWrapper isFeatured={isFeatured}>
      {isFeaturedLoading ? (
        <Loader className={className} size={18} />
      ) : (
        <FeatureHeartStyled
          className={className}
          onClick={() => {
            if (isFeatured) {
              removeFeaturedPOAP(poapTokenId);
            } else {
              addFeaturedPOAP(poapTokenId);
            }
          }}
        />
      )}
    </HeartWrapper>
  );
};
