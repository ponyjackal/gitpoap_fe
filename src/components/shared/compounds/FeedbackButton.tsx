import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { Popover, Box } from '@mantine/core';
import { useDisclosure } from '@mantine/hooks';
import { GitPOAPLogoNoText } from '../elements/icons';
import { BackgroundPanel, BackgroundPanel2, ExtraHover, ExtraPressed } from '../../../colors';
import { Text } from '../elements';

const StyledLogo = styled(GitPOAPLogoNoText)`
  height: ${rem(45)};
  width: ${rem(45)};
  transition: transform 200ms ease-in-out;
  path {
    transition: fill 200ms ease-in-out;
  }
`;

const CircleButton = styled.a`
  display: flex;
  align-items: center;
  justify-content: center;
  height: ${rem(70)};
  width: ${rem(70)};
  background-color: ${BackgroundPanel};
  transition: background-color 200ms ease-in-out, transform 200ms ease-in-out;
  border-radius: 50%;
  border: ${rem(3)} solid ${BackgroundPanel2};
  &:hover {
    background-color: ${BackgroundPanel2};
    transform: scale(1.1);
    cursor: pointer;
    ${StyledLogo} {
      transform: scale(1.1);
      path {
        fill: ${ExtraHover};
      }
    }
  }
  &:active {
    background-color: ${BackgroundPanel};
    ${StyledLogo} {
      color: ${ExtraHover};
      path {
        fill: ${ExtraPressed};
      }
    }
  }
`;

const PopoverContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

type Props = {
  className?: string;
  href: string;
};

export const FeedbackButton = ({ className, href }: Props) => {
  const [isOpen, { open, close }] = useDisclosure(false);
  return (
    <Popover
      opened={isOpen}
      onClose={close}
      position="top"
      withArrow
      trapFocus={false}
      closeOnEscape={false}
      transition="fade"
      transitionDuration={200}
      radius="lg"
      withinPortal
      zIndex={100}
      styles={{
        dropdown: {
          backgroundColor: BackgroundPanel,
        },
      }}
    >
      <Popover.Target>
        <Box onMouseEnter={open} onMouseLeave={close} className={className} style={{ zIndex: 100 }}>
          <CircleButton href={href} target="_blank" rel="noopener noreferrer">
            <StyledLogo />
          </CircleButton>
        </Box>
      </Popover.Target>
      <Popover.Dropdown>
        <PopoverContainer>
          <Text>{'Give Feedback'}</Text>
        </PopoverContainer>
      </Popover.Dropdown>
    </Popover>
  );
};
