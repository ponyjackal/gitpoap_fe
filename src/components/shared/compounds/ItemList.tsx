import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { Header } from '../elements/Header';
import { Button } from '../elements/Button';
import { FaPlus } from 'react-icons/fa';
import { Select } from '../elements/Select';
import { Text } from '../elements/Text';
import { TextGray } from '../../../colors';
import { Input } from '../elements/Input';

type Props = {
  title?: string;
  className?: string;
  children?: React.ReactNode;
  selectOptions: SelectOption[];
  selectValue: string;
  onSelectChange: (value: string) => void;
  isLoading: boolean;
  hasShowMoreButton: boolean;
  showMoreOnClick: () => void;
  searchInputPlaceholder?: string;
  searchInputValue?: string;
  onSearchInputChange?: React.ChangeEventHandler<HTMLInputElement>;
};

export type SelectOption<T = string> = { value: T; label: string };

const Container = styled.div`
  display: inline-flex;
  flex-direction: column;
  width: 100%;
`;

const Heading = styled.div`
  display: inline-flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const ListTitle = styled(Header)`
  font-size: ${rem(30)};
  line-height: ${rem(42)};
`;

const Sorting = styled.div`
  display: inline-flex;
  flex-direction: row;
`;

const ShowMore = styled(Button)`
  align-self: center;
`;

const SortBy = styled(Text)`
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: ${rem(12)};
  line-height: ${rem(18)};
  letter-spacing: ${rem(2)};
  text-transform: uppercase;
  color: ${TextGray};
  margin-right: ${rem(10)};
`;

const SearchInput = styled(Input)`
  margin-right: ${rem(50)};
`;

export const ItemList = ({
  children,
  className,
  title,
  selectOptions,
  selectValue,
  onSelectChange,
  isLoading,
  hasShowMoreButton,
  showMoreOnClick,
  searchInputPlaceholder,
  searchInputValue,
  onSearchInputChange,
}: Props) => {
  return (
    <Container className={className}>
      <Heading>
        <ListTitle>{title}</ListTitle>
        <Sorting>
          {searchInputValue !== undefined && onSearchInputChange && (
            <SearchInput
              placeholder={searchInputPlaceholder}
              value={searchInputValue}
              onChange={onSearchInputChange}
            />
          )}
          <SortBy>{'Sort By: '}</SortBy>
          <Select data={selectOptions} value={selectValue} onChange={onSelectChange} />
        </Sorting>
      </Heading>
      {children}
      {hasShowMoreButton && (
        <ShowMore
          onClick={showMoreOnClick}
          leftIcon={<FaPlus />}
          variant="outline"
          loading={isLoading}
        >
          {'Show more'}
        </ShowMore>
      )}
    </Container>
  );
};
