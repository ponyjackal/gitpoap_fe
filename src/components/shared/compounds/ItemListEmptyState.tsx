import React from 'react';
import { rem } from 'polished';
import { Group } from '@mantine/core';

type Props = {
  icon?: React.ReactNode;
  children: React.ReactNode;
};

export const EmptyState = ({ children, icon }: Props) => {
  return (
    <Group
      direction="column"
      position="center"
      align="center"
      grow
      style={{ padding: rem(60), flex: '1', gridColumnEnd: 1, gridRowEnd: 2 }}
    >
      {icon && icon}
      {children}
    </Group>
  );
};
