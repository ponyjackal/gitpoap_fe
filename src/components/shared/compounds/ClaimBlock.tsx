import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { Button } from '../elements/Button';
import { GitPOAP } from './GitPOAP';
import { FaCheckCircle, FaCoins, FaEthereum } from 'react-icons/fa';

type Props = {
  gitPOAPId: number;
  imgSrc: string;
  name: string;
  orgName: string;
  description: string;
  onClickClaim: () => void;
  onClickBadge?: () => void;
  isClaimed?: boolean;
  isLoading?: boolean;
  isConnected?: boolean;
};

const Wrapper = styled.div`
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  height: ${rem(340)};
`;

const ButtonWrapper = styled.div`
  flex: 1;
  margin-top: ${rem(14)};
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
`;

const getButtonText = (isClaimed: boolean | undefined, isConnected: boolean | undefined) => {
  if (isClaimed) {
    return 'Minted';
  }

  if (!isConnected) {
    return 'Connect';
  }

  return 'Mint';
};

export const ClaimBlock = ({
  gitPOAPId,
  imgSrc,
  name,
  orgName,
  description,
  onClickClaim,
  onClickBadge,
  isClaimed,
  isLoading,
  isConnected,
}: Props) => {
  return (
    <Wrapper>
      <GitPOAP
        gitPOAPId={gitPOAPId}
        imgSrc={imgSrc}
        name={name}
        orgName={orgName}
        description={description}
        onClick={onClickBadge}
      />
      <ButtonWrapper>
        <Button
          onClick={onClickClaim}
          loading={isLoading}
          leftIcon={isClaimed ? <FaCheckCircle /> : !isConnected ? <FaEthereum /> : <FaCoins />}
          disabled={isClaimed}
        >
          {getButtonText(isClaimed, isConnected)}
        </Button>
      </ButtonWrapper>
    </Wrapper>
  );
};
