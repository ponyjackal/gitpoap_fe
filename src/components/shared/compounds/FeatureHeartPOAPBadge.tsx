import React from 'react';
import { Loader } from '../elements/Loader';
import { useFeaturedPOAPsDispatch } from '../../profile/FeaturedPOAPsContext';
import { HeartWrapper, FeatureHeartStyled } from './FeatureHeart';

type Props = {
  className?: string;
  poapTokenId: string;
  isFeatured: boolean;
  isFeaturedLoading: boolean;
};

export const FeaturedHeartPOAPBadge = ({
  className,
  poapTokenId,
  isFeatured,
  isFeaturedLoading,
}: Props) => {
  const { addFeaturedPOAP, removeFeaturedPOAP } = useFeaturedPOAPsDispatch();
  return (
    <HeartWrapper isFeatured={isFeatured}>
      {isFeaturedLoading ? (
        <Loader className={className} size={18} />
      ) : (
        <FeatureHeartStyled
          className={className}
          onClick={() => {
            if (isFeatured) {
              removeFeaturedPOAP(poapTokenId);
            } else {
              addFeaturedPOAP(poapTokenId);
            }
          }}
        />
      )}
    </HeartWrapper>
  );
};
