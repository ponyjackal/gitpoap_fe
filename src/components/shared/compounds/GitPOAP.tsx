import { Link } from '../../Link';
import React from 'react';
import styled, { css } from 'styled-components';
import { rem } from 'polished';
import { GitPOAPBadge } from '../elements/GitPOAPBadge';
import { Title } from '../elements/Title';
import { TextLight } from '../../../colors';
import { FeatureHeart } from './FeatureHeart';

type Props = {
  gitPOAPId: number;
  imgSrc: string;
  name: string;
  orgName: string;
  description?: string;
  className?: string;
  poapTokenId?: string;
  onClick?: () => void;
};

const LineClamp = (lines: number) => css`
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: ${lines};
  -webkit-box-orient: vertical;
`;

const Wrapper = styled.div`
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const Info = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: ${rem(150)};
`;

const TitleStyled = styled(Title)`
  margin-top: ${rem(10)};
  text-align: center;
  ${LineClamp(3)};
`;

const OrgName = styled.div`
  font-family: PT Mono;
  font-style: normal;
  font-weight: bold;
  font-size: ${rem(11)};
  line-height: ${rem(18)};
  text-align: center;
  letter-spacing: ${rem(1.2)};
  text-transform: uppercase;
  color: ${TextLight};
  margin-top: ${rem(8)};
`;

const Description = styled.div`
  font-family: PT Mono;
  font-style: normal;
  font-weight: normal;
  font-size: ${rem(11)};
  line-height: ${rem(14)};
  text-align: center;
  letter-spacing: ${rem(-0.1)};
  color: ${TextLight};
  margin-top: ${rem(8)};
  ${LineClamp(2)};
`;

const Heart = styled(FeatureHeart)`
  position: absolute;
  bottom: ${rem(0)};
  right: ${rem(10)};
`;

const BadgeWrapper = styled(Wrapper)`
  position: relative;
`;

export const GitPOAP = ({
  className,
  poapTokenId,
  gitPOAPId,
  imgSrc,
  name,
  orgName,
  description,
  onClick,
}: Props) => {
  return (
    <Wrapper className={className}>
      <BadgeWrapper>
        <Link href={`/gp/${gitPOAPId}`} passHref>
          <GitPOAPBadge
            size="sm"
            imgUrl={imgSrc}
            onClick={() => {
              if (onClick) {
                onClick();
              }
            }}
          />
          {poapTokenId && <Heart poapTokenId={poapTokenId} />}
        </Link>
      </BadgeWrapper>
      <Info>
        <Link href={`/gp/${gitPOAPId}`} passHref>
          <TitleStyled>{name}</TitleStyled>
        </Link>
        <OrgName>{orgName}</OrgName>
        {description && <Description>{description}</Description>}
      </Info>
    </Wrapper>
  );
};
