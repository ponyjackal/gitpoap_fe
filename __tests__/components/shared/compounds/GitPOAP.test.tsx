import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { GitPOAP } from '../../../../src/components/shared/compounds/GitPOAP';

describe('GitPOAP', () => {
  it('should render with name', () => {
    render(
      <GitPOAP
        gitPOAPId={1}
        imgSrc="https://poap.gallery/static/media/test_gitpoap.1b1b1b1b.png"
        name={'GitPOAP: 2017 OpenZeppelin Contracts Contributor'}
        repoName={'testRepo'}
        orgName={'Polygon'}
        description={'To the creators of Polygon Network'}
      />,
    );
    const nameText = screen.getByText('2017 OpenZeppelin Contracts Contributor');
    expect(nameText).toBeInTheDocument();
    expect(nameText).toMatchSnapshot();
  });

  it('should render with description', () => {
    render(
      <GitPOAP
        gitPOAPId={1}
        imgSrc="https://poap.gallery/static/media/test_gitpoap.1b1b1b1b.png"
        name={'GitPOAP: 2017 OpenZeppelin Contracts Contributor'}
        repoName={'testRepo'}
        orgName={'Polygon'}
        description={'To the creators of Polygon Network'}
      />,
    );
    const descriptionText = screen.getByText('To the creators of Polygon Network');
    expect(descriptionText).toBeInTheDocument();
    expect(descriptionText).toMatchSnapshot();
  });

  it('should render with repoName and orgName', () => {
    render(
      <GitPOAP
        gitPOAPId={1}
        imgSrc="https://poap.gallery/static/media/test_gitpoap.1b1b1b1b.png"
        name={'GitPOAP: 2017 OpenZeppelin Contracts Contributor'}
        repoName={'testRepo'}
        orgName={'Polygon'}
        description={'To the creators of Polygon Network'}
      />,
    );
    const repoNameLink = screen.getByRole('link', { name: 'testRepo' });
    expect(repoNameLink).toBeInTheDocument();
    expect(repoNameLink).toHaveAttribute('href', '/gh/Polygon/testRepo');
    expect(repoNameLink).toMatchSnapshot();
  });

  it('should render with repoId', () => {
    render(
      <GitPOAP
        gitPOAPId={1}
        imgSrc="https://poap.gallery/static/media/test_gitpoap.1b1b1b1b.png"
        name={'GitPOAP: 2017 OpenZeppelin Contracts Contributor'}
        repoName={'testRepo'}
        repoId={22}
        description={'To the creators of Polygon Network'}
      />,
    );
    const repoNameLink = screen.getByRole('link', { name: 'testRepo' });
    expect(repoNameLink).toBeInTheDocument();
    expect(repoNameLink).toHaveAttribute('href', '/rp/22');
    expect(repoNameLink).toMatchSnapshot();
  });

  it('should render without repoId and orgName', () => {
    render(
      <GitPOAP
        gitPOAPId={1}
        imgSrc="https://poap.gallery/static/media/test_gitpoap.1b1b1b1b.png"
        name={'GitPOAP: 2017 OpenZeppelin Contracts Contributor'}
        repoName={'testRepo'}
        description={'To the creators of Polygon Network'}
      />,
    );
    const repoNameText = screen.getByText('testRepo');
    expect(repoNameText).toBeInTheDocument();

    const repoNameLink = screen.queryByRole('link', { name: 'testRepo' });
    expect(repoNameLink).not.toBeInTheDocument();
    expect(repoNameLink).toMatchSnapshot();
  });
});
