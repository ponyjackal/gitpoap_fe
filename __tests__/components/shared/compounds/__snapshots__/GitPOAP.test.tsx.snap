// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitPOAP should render with description 1`] = `
.c0 {
  font-family: PT Mono;
  font-style: normal;
  font-weight: normal;
  font-size: 0.6875rem;
  line-height: 0.875rem;
  text-align: center;
  -webkit-letter-spacing: -0.00625rem;
  -moz-letter-spacing: -0.00625rem;
  -ms-letter-spacing: -0.00625rem;
  letter-spacing: -0.00625rem;
  color: #E2E2EE;
  margin-top: 0.5rem;
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

<div
  class="c0"
>
  To the creators of Polygon Network
</div>
`;

exports[`GitPOAP should render with name 1`] = `
.c0 {
  display: inline-block;
  font-family: 'PT Mono',monospace;
  font-weight: 700;
  font-size: 0.8125rem;
  line-height: 1.0625rem;
  -webkit-letter-spacing: 0.03125rem;
  -moz-letter-spacing: 0.03125rem;
  -ms-letter-spacing: 0.03125rem;
  letter-spacing: 0.03125rem;
  text-align: center;
  color: #307AE8;
  -webkit-transition: color 150ms ease-in-out;
  transition: color 150ms ease-in-out;
  cursor: pointer;
  color: #307AE8;
}

.c1:hover:not([disabled]) {
  color: #5596F6;
}

.c1:active:not([disabled]) {
  color: #1761CF;
}

.c1[disabled] {
  color: #393A51;
  cursor: not-allowed;
}

.c2 {
  margin-top: 0.625rem;
  text-align: center;
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
}

.emotion-1 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1:focus {
  outline-offset: 2px;
  outline: 2px solid #339af0;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 c0 c1 c2 emotion-1"
>
  2017 OpenZeppelin Contracts Contributor
</div>
`;

exports[`GitPOAP should render with repoId 1`] = `
.c0 {
  font-family: PT Mono;
  font-style: normal;
  font-weight: bold;
  font-size: 0.6875rem;
  line-height: 1.125rem;
  text-align: center;
  -webkit-letter-spacing: 0.075rem;
  -moz-letter-spacing: 0.075rem;
  -ms-letter-spacing: 0.075rem;
  letter-spacing: 0.075rem;
  text-transform: uppercase;
  color: #E2E2EE;
  margin-top: 0.5rem;
  -webkit-transition: color 200ms ease-in-out;
  transition: color 200ms ease-in-out;
}

.c0:hover {
  color: #5596F6;
}

.c0:active {
  color: #1761CF;
}

<a
  href="/rp/22"
>
  <div
    class="c0"
  >
    testRepo
  </div>
</a>
`;

exports[`GitPOAP should render with repoName and orgName 1`] = `
.c0 {
  font-family: PT Mono;
  font-style: normal;
  font-weight: bold;
  font-size: 0.6875rem;
  line-height: 1.125rem;
  text-align: center;
  -webkit-letter-spacing: 0.075rem;
  -moz-letter-spacing: 0.075rem;
  -ms-letter-spacing: 0.075rem;
  letter-spacing: 0.075rem;
  text-transform: uppercase;
  color: #E2E2EE;
  margin-top: 0.5rem;
  -webkit-transition: color 200ms ease-in-out;
  transition: color 200ms ease-in-out;
}

.c0:hover {
  color: #5596F6;
}

.c0:active {
  color: #1761CF;
}

<a
  href="/gh/Polygon/testRepo"
>
  <div
    class="c0"
  >
    testRepo
  </div>
</a>
`;

exports[`GitPOAP should render without repoId and orgName 1`] = `null`;
